### ISSUE TITLE
Add walkthrough of cycling background colours

### ISSUE THREAD

**Issue by gurcei** [MEGA65/mega65-user-guide#118]
I had this document walking through the background colour cycling example, comparing the way we remember it in the c64 days, compared against the differences experienced in the c65/mega65 world:

https://docs.google.com/document/d/1UrpX10ZZD2qJz2aV0O4V-dDNjhRDJkGWEfLxOJ5abpg/edit?usp=sharing

Paul suggested putting it in the developer's guide.

I'm not ready to tackle the latex learning curve as yet, so Paul suggester creating a ticket here, in-case others more latex savvy could port it across.

**Comment by isedwards** [comment]
The original BASIC program was very simple.

Was it possible to see the machine code that the BASIC interpreter created and then to work all of this out from there?

**Comment by Edilbert** [comment]
I can take care of it, see my reply on discord @gurcei 
@isedwards , an interpreter does not create machine code, it interprets and executes.
A BASIC compiler would generate code, but typically not assembly code, but in most cases another code, that has to be interpreted. Well known is the P-Code, that was first used with Pascal compilers.
AFAIK, there are no BASIC compilers for the MEGA65 now, but a C compiler.
The C compiler however creates machine language instructions, which may be looked at and hand optimised.

**Comment by isedwards** [comment]
Thank you for the explanation @Edilbert

In case it's helpful, I replied on discord:
> I'd be happy to convert it from it's current form straight to LaTeX and images if it's helpful? Just let me know where you would want it added.

**Comment by gardners** [comment]
Thank you for offering to tackle this!

I suggest starting by putting it in a separate latex file, examples-colourcycling.tex which you can \input from the sandbox after \begin{appendices} for testing. Then when you are happy with it, we can put it into the right place in the book(s).

**Comment by gurcei** [comment]
Ok, as I've acquired a bit more latex smarts since I last pondered this topic, I'm willing to try add this text into a section of the developer guide in future.

Since at present, we're focusing on the user-guide, I'll leave this task on the back-burner until all user-guide matters are sorted.

**Comment by dansanderson** [comment]
I enjoyed reading this early article! With appreciation and respect, I'm going to resolve this issue, for a few reasons:

* The motivating example no longer applies. Later ROMs use a different SYS MAP that doesn't need to be changed for the two-line assembly program at $2005 to be reachable by SYS. BANK 0 is no longer needed, either.
* At some point we corrected the relationship between MAP and the D030/01 banking methods. (I assume this is why this example zeroes out the MAP? I'm unsure.)
* Memory mapping concepts are now described in the Memory chapter.

The article might still be of historical interest if it applies to the C65 prototype ROM. Maybe worth adapting into a blog post?
